src/Nethermind/Nethermind.Evm/Tracing/GethStyle/Custom/Native/GethLikeBlockNativeTracer.cs

```diff
@@ -0,0 +1,24 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+namespace Nethermind.Evm.Tracing.GethStyle.Custom.Native;
+
+public class GethLikeBlockNativeTracer : BlockTracerBase<GethLikeTxTrace, GethLikeNativeTxTracer>
+{
+    private readonly GethTraceOptions _options;
+
+    public GethLikeBlockNativeTracer(GethTraceOptions options) : base(options.TxHash)
+    {
+        _options = options;
+    }
+
+    protected override GethLikeNativeTxTracer OnStart(Transaction? tx)
+    {
+        return GethLikeNativeTracerFactory.CreateTracer(_options);
+    }
+
+    protected override bool ShouldTraceTx(Transaction? tx) => base.ShouldTraceTx(tx) && tx is not null;
+
+    protected override GethLikeTxTrace OnEnd(GethLikeNativeTxTracer txTracer) => txTracer.BuildResult();
```

Comment:
If we potentially introduce one more class between `GethLikeBlockNativeTracer` and `BlockTracerBase`

like: 

`abstract class GethLikeBlockTracer<TTxTracer, TEntry>: BlockTracerBase<GethLikeTxTrace, TTxTracer> where TTxTracer : GethLikeTxTracer<TEntry> where TEntry : GethTxTraceEntry, new()`

We could remove those 2 method overrides in those end-tracers, not sure if worth it

Reply:
I spiked it out but ultimately felt like it made the code more convoluted. Unless you feel strongly about this change I'm going to leave it out